frontend/src/Component/EditComponents/EditComponents.tsx

```diff
@@ -33,36 +44,39 @@ export default function EditComponents(props: EditComponentsProp) {
                     flexDirection: 'column',
                 }}>
                     <TableContainer component={Paper}>
-                        <Table aria-label="simple table">
-                            <TableHeadComponentTable cells={[{cellName:"Component", align: undefined},
-                                {cellName:"Model", align:"left"}, {cellName:"Age (km)", align:"right"},
-                                {cellName:"", align: "right"}]}/>
-                            <TableBody>
-                                {props.components.map((component) => (
-                                    <TableRow
-                                        key={component.category}
-                                        sx={{ '&:last-child td, &:last-child th': { border: 0 } }}
-                                    >
-                                        <TableCell component="th" scope="row">
-                                            {component.category}
-                                        </TableCell>
-                                        <TableCell align="left">{component.type}</TableCell>
-                                        <TableCell align="right">{component.age}</TableCell>
-                                        <TableCell align="right" sx={{
-                                            p: 0,
-                                            width: 20
-                                        }}>
-                                            <DeleteIcon onClick={() =>handleDeleteComponent(component)} sx={{
-                                                alignSelf: 'end',
-                                                cursor: 'pointer',
-                                                color: '#2196f3',
-                                                mr: 1
-                                            }}/>
-                                        </TableCell>
-                                    </TableRow>
-                                ))}
-                            </TableBody>
-                        </Table>
+                        {props.components.length>0?
```

Comment:
It's a bit hard to read. Maybe you could put the code from lines 48-77 into a constant so it is easier to see what your two possible outputs are?